File: FlankSpeedLegacyDemo.Web/Models/AuditLog.cs
namespace FlankSpeedLegacyDemo.Web.Models;

public class AuditLog
{
    public int Id { get; set; }
    public int WorkItemId { get; set; }
    public WorkItem? WorkItem { get; set; }

    // e.g., "Created", "Status Changed", "Assigned", "Comment Added"
    public string Action { get; set; } = string.Empty;

    public int PerformedByUserId { get; set; }
    public User? PerformedByUser { get; set; }
    public string? OldValue { get; set; }
    public string? NewValue { get; set; }
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
}

File: power-apps-migration/src/models/AuditLog.ts
/**
 * Migrated from: FlankSpeedLegacyDemo.Web/Models/AuditLog.cs
 * Source: Audit trail for all WorkItem changes
 * Target: Azure SQL [dbo].[AuditLogs] via Power Platform SQL connector
 *
 * Migration note: In the Power Apps target, audit logging can be handled by
 * Power Automate flows triggered on data changes, or via application-level
 * logging in the TypeScript service layer.
 */
import { User } from "./User";
import { WorkItem } from "./WorkItem";

export interface AuditLog {
  id: number;
  workItemId: number;
  workItem?: WorkItem;
  /** e.g., "Created", "Status Changed", "Assigned", "Comment Added" */
  action: string;
  performedByUserId: number;
  performedByUser?: User;
  oldValue: string | null;
  newValue: string | null;
  timestamp: string;
}
